### PR TITLE
Species Simplifier™ - Part 1

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
First part of the PR to make mon easier to add.

- Grouped `gSpeciesInfo` by species family.
- Added Mega Evolution hidden ability failsafe.
- Converted `P_UPDATED_STATS` "ifdef blocks" to ternaries.
- Fixed Mega Gardevoir/Gallade updated Egg Groups.
- Fixed mon not having their appropiate items
	- Alolan Sandlash (Grip Claw, Rare)
	- Alolan Muk (Black Sludge, Common)
	- Alolan Golem (Cell Battery, Common)
	- Alolan Marowak (Thick Club, Rare)
	- Alolan Ninetales (Snowball, Rare)
- Fixed mon not having the Legendary flag:
	- Giratina Origin
- Fixed mon not having the Mythical flag:
	- Deoxys Attack
	- Deoxys Defense
	- Deoxys Speed
	- Shaymin Sky
- Fixed mon having inproper `noFlip` property.
	- Galarian Weezing
- Fixed Aegislash Blade giving the wrong EV yield.
- Fixed Zygarde 10% with Power Construct having the same stats as Zygarde 50%.

## **Discord contact info**
AsparagusEduardo

### CODE REVIEW

File: src/data/pokemon/form_species_table_pointers.h [hunk 1]
```diff
@@ -684,8 +684,12 @@ const u16 *const gFormSpeciesIdTables[NUM_SPECIES] =
     [SPECIES_LAPRAS_GIGANTAMAX] = sLaprasFormSpeciesIdTable,
     [SPECIES_EEVEE_GIGANTAMAX] = sEeveeFormSpeciesIdTable,
     [SPECIES_SNORLAX_GIGANTAMAX] = sSnorlaxFormSpeciesIdTable,
+#if P_GEN_5_POKEMON == TRUE
     [SPECIES_GARBODOR_GIGANTAMAX] = sGarbodorFormSpeciesIdTable,
+#endif
+#if P_GEN_7_POKEMON == TRUE
     [SPECIES_MELMETAL_GIGANTAMAX] = sMelmetalFormSpeciesIdTable,
+#endif
     [SPECIES_RILLABOOM_GIGANTAMAX] = sRillaboomFormSpeciesIdTable,
```
Comment (DizzyEggg) [inline src/data/pokemon/form_species_table_pointers.h:693]: I think it's missing #if P_GEN_8_POKEMON == TRUE

**Review by DizzyEggg — CHANGES_REQUESTED** [review]
Why are the Gmax pokemon even in the gSpeciesInfo? They have the same data as regular mons?

**Comment by AsparagusEduardo** [comment]
> Why are the Gmax pokemon even in the gSpeciesInfo? They have the same data as regular mons?

Currently, it was easier to add them as separate species so that they use the separate graphics and use the form-change system.
We could in the future have a separate table for Gmax mon graphics and reduce the space used in `gSpeciesInfo`

File: src/data/pokemon/form_species_table_pointers.h (hunk 1 above)
Comment (AsparagusEduardo) [reply src/data/pokemon/form_species_table_pointers.h:693]: Oh, apparently all Gmax forms are surrounded by `#if P_GEN_8_POKEMON == TRUE`.
I'll change that so it only groups the appropiate ones.

File: src/data/pokemon/teachable_learnset_pointers.h [hunk 2]
```diff
@@ -1540,8 +1540,12 @@ const u16 *const gTeachableLearnsets[NUM_SPECIES] =
     [SPECIES_LAPRAS_GIGANTAMAX] = sLaprasTeachableLearnset,
     [SPECIES_EEVEE_GIGANTAMAX] = sEeveeTeachableLearnset,
     [SPECIES_SNORLAX_GIGANTAMAX] = sSnorlaxTeachableLearnset,
+#if P_GEN_5_POKEMON == TRUE
     [SPECIES_GARBODOR_GIGANTAMAX] = sGarbodorTeachableLearnset,
+#endif
+#if P_GEN_7_POKEMON == TRUE
     [SPECIES_MELMETAL_GIGANTAMAX] = sMelmetalTeachableLearnset,
+#endif
     [SPECIES_RILLABOOM_GIGANTAMAX] = sRillaboomTeachableLearnset,
```
[the remark this answers is not in the document]
Comment (AsparagusEduardo) [reply src/data/pokemon/teachable_learnset_pointers.h:1549]: Same as above.

File: src/data/pokemon/form_species_table_pointers.h (hunk 1 above)
Comment (AsparagusEduardo) [reply src/data/pokemon/form_species_table_pointers.h:693]: Actually, I'm gonna move these to `gSpeciesInfo` anyway, and those are already grouped by family. Is it ok to skip this for now?

**Review by DizzyEggg — APPROVED** [review]

**Pull request merged**